From c2b2fcb163e7cf8061b2f3311f861dc93992980e Mon Sep 17 00:00:00 2001
From: Theo Linkspfeifer <lastonestanding@tutanota.com>
Date: Sat, 11 Jan 2025 00:02:03 +0100
Subject: [PATCH] Prevent files from vanishing after save (#1535)

This as well fixes "select after rename" for slow devices (See #1415)

Fixes #1535
---
 thunar/thunar-folder.c | 10 ++--------
 1 file changed, 2 insertions(+), 8 deletions(-)

diff --git a/thunar/thunar-folder.c b/thunar/thunar-folder.c
index c7c3941f5..52842c8f4 100644
--- thunar/thunar-folder.c
+++ thunar/thunar-folder.c
@@ -1039,16 +1039,10 @@ thunar_folder_monitor (GFileMonitor     *monitor,
         renamed_file = event_file_thunar;
 
       /* remove both files from the internal maps */
-      /* trigger the timeout manually, so that connected models will get updated (required for move+replace use-case) */
       if (event_file_thunar != NULL)
-        {
-          thunar_folder_remove_file (folder, event_file_thunar);
-          _thunar_folder_files_update_timeout (folder);
-        }
+        thunar_folder_remove_file (folder, event_file_thunar);
       if (other_file_thunar != NULL)
-        {
-          thunar_folder_remove_file (folder, other_file_thunar);
-        }
+        thunar_folder_remove_file (folder, other_file_thunar);
 
       /* replace the GFile in the ThunarFile for the renamed file with the new gfile */
       thunar_file_replace_file (renamed_file, other_file);
-- 
GitLab

